### PR TITLE
Make the api endpoints timezone-aware

### PR DESCRIPTION
**Pull request by eswan18**
## Summary
- Implements timezone-aware date filtering throughout the API to ensure users can interact with data in their local timezone
- Maintains UTC standardization internally while providing local timezone support for user-facing operations
- Adds comprehensive timezone conversion utilities and updates all date-filtering aggregation functions

### 🌐 New Timezone Infrastructure
- **New utility module**: `fitness/utils/timezone.py` with core conversion functions
- **UTC to local conversion**: Converts UTC dates to user's local dates for proper bucketing
- **Timezone-aware filtering**: Filters runs by local date ranges instead of UTC
- **Bulk conversion helpers**: Efficient timezone conversion for lists of runs

### 📊 Updated Aggregation Functions
All date-filtering functions now support optional `user_timezone` parameter:
- **Mileage**: `total_mileage()`, `avg_miles_per_day()`, `miles_by_day()`, `rolling_sum()`
- **Training Load**: `training_stress_balance()`, `trimp_by_day()`
- **Duration**: `total_seconds()`

### 🔌 API Endpoint Updates
- **All `/metrics/*` endpoints**: Added `user_timezone` parameter
- **`/runs` endpoint**: Supports timezone-aware date filtering
- **Backward compatible**: Defaults to UTC when timezone not specified

### 🧪 Comprehensive Testing
- **17 new test cases** covering timezone conversion logic
- **Edge cases tested**: DST transitions, year boundaries, invalid timezones
- **Multiple timezone scenarios**: Chicago, Tokyo, Honolulu, New York, London, Sydney
- **Error handling validation**: Invalid timezone names, empty data sets

## Benefits for Users
- ✅ **Local date filtering**: Users can filter by local dates (e.g., "show runs from Jan 15-16 Chicago time")
- ✅ **Proper daily bucketing**: Daily aggregations bucket by user's local dates, not UTC
- ✅ **Training load accuracy**: Training stress calculations respect user's day boundaries
- ✅ **Consistent data**: Uniform date assignment across historical (MMF) and current (Strava) data

## Technical Implementation
- **UTC internal storage**: All data remains UTC internally for consistency
- **Conversion at aggregation layer**: Timezone conversion applied only when computing results
- **Clean separation of concerns**: Timezone logic isolated in utility functions
- **Efficient processing**: Minimal overhead for existing UTC-only usage

## Test Plan
- [x] All existing tests pass (backward compatibility verified)
- [x] New timezone tests cover edge cases and error scenarios
- [x] Manual testing with different timezone scenarios
- [x] Performance impact verified to be minimal

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: api/fitness/utils/timezone.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
+"""Timezone utility functions for converting between UTC and user timezones."""
+
+from datetime import date, datetime, timezone
+import zoneinfo
```
[the remark this answers is not in the document]
Comment (eswan18) [reply api/fitness/utils/timezone.py:4]: no, this app doesn't run on python 3.9 🤦

File: api/fitness/load/mmf/load.py [hunk 2]
```diff
@@ -1,26 +1,51 @@
...
+            activity = MmfActivity.model_validate(row)
+            # Convert the workout_date from local timezone to UTC
+            activity.workout_date_utc = _convert_date_to_utc(activity.workout_date, tz)
+            records.append(activity)
     return records
 
 
-def load_mmf_runs(mmf_file: Path | None = None) -> list[MmfActivity]:
+def _convert_date_to_utc(local_date, local_tz: zoneinfo.ZoneInfo):
+    """Convert a naive date to UTC by assuming it represents the start of day in local_tz."""
```
[the remark this answers is not in the document]
Comment (eswan18) [reply api/fitness/load/mmf/load.py:35]: meh, I'll do this another time.

File: api/fitness/load/mmf/load.py [hunk 3]
```diff
@@ -1,26 +1,51 @@
...
+        for row in reader:
+            activity = MmfActivity.model_validate(row)
+            # Convert the workout_date from local timezone to UTC
+            activity.workout_date_utc = _convert_date_to_utc(activity.workout_date, tz)
+            records.append(activity)
     return records
 
 
-def load_mmf_runs(mmf_file: Path | None = None) -> list[MmfActivity]:
+def _convert_date_to_utc(local_date, local_tz: zoneinfo.ZoneInfo):
```
[the remark this answers is not in the document]
Comment (eswan18) [reply api/fitness/load/mmf/load.py]: fixed in 8b868bd

File: api/tests/utils/test_timezone.py [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+        """Test timezone conversion that crosses year boundary backwards."""
+        # New Year's Day UTC becomes New Year's Eve in western timezones
+        utc_date = date(2025, 1, 1)
+        result = convert_utc_date_to_user_timezone(utc_date, "Pacific/Honolulu")
+        # Should handle year boundary correctly
+        assert result == date(2024, 12, 31)  # UTC midnight = 2 PM previous day in Honolulu
+
+    def test_invalid_timezone_raises_error(self):
+        """Test that invalid timezone raises appropriate error."""
+        with pytest.raises(Exception):  # Should raise ZoneInfoNotFoundError or similar
```
[the remark this answers is not in the document]
Comment (eswan18) [reply api/tests/utils/test_timezone.py]: fixed in 7a3d125
